### ISSUE TITLE
Found a possible parsing error:

### ISSUE THREAD

**Issue by manusimidt** [manusimidt/xbrl_parser#40]
Found a possible parsing error:

https://www.sec.gov/ix?doc=/Archives/edgar/data/0000320193/000032019320000096/aapl-20200926.htm

**CommonStockParOrStatedValuePerShare**  should have 2 instances, but the lib only gives back 09/28/2019, even when not filtering for len(fact.context.segments)

Common Stock, Par or Stated Value Per Share
As of 09/26/2020
0.00001

Common Stock, Par or Stated Value Per Share
As of 09/28/2019
0.00001

Same problem for
- CommonStockSharesAuthorized  (missing As of 09/26/2020)
- EffectiveIncomeTaxRateReconciliationAtFederalStatutoryIncomeTaxRate (missing 12 months ending 09/28/2019)

_Originally posted by @mrx23dot in https://github.com/manusimidt/xbrl_parser/discussions/36#discussioncomment-938050_

**Comment by manusimidt** [comment]
You are right, in this case the number is tagged twice by two encapsulating facts:

```xml
<ix:nonFraction 
  unitRef="usdPerShare" 
  contextRef="i747bec89b4e84f74ae3445db3509f609_I20200926" 
  decimals="INF" name="us-gaap:CommonStockParOrStatedValuePerShare" 
  scale="0"
  id="...">

  <ix:nonFraction 
    unitRef="usdPerShare" 
    contextRef="ic363a9ee75bd48f88800fc5dadba8293_I20190928" 
    decimals="INF" 
    name="us-gaap:CommonStockParOrStatedValuePerShare" 
    scale="0" id="...">
       0.00001
  </ix:nonFraction>
</ix:nonFraction> 
```

I will check the inline XBRL Specificiation and think about how the parser could implement this.

**Comment by manusimidt** [comment]
The current inline XBRL Specification 1.1 does not mention the possibility of tagging one number twice. 🤔
https://www.xbrl.org/specification/inlinexbrl-part1/rec-2013-11-18/inlinexbrl-part1-rec-2013-11-18.html#d1e5896
 
However, since this seems to occur in multiple submissions and doesn't conflict with other elements of the submission, it would be good to just implement this anyway.

**Comment by rayniervanegmond** [comment]
In the inlineXBR specification it says:

"The ix:nonFraction element MUST have exactly one child which SHALL be either an ix:nonFraction element or a text node, unless it has an xsi:nil attribute with the value true."

This means that nesting can occur -- I wouldn't know why one would do it but the spec allows it. BTW as the spec declares, it can have no child elements if the "xsi:nil attribute" of the ix:nonFraction element is True. Funny enough this could also mean that a "nested-list" is a possibility also. 

All that beind said, nesting ix:nonFraction elements is very uncommon.

**Comment by manusimidt** [comment]
Thanks for your comment! Yes, I agree, nesting `nonFraction`-facts is very uncommon. What I had seen more often was nested `nonNumeric` facts, like the following example:
```xml
<ix:nonNumeric
  	name="dei:DocumentPeriodEndDate"
        format="ixt:datemonthdayyearen">
  	  March 31,
  	<ix:nonNumeric
  		name="dei:DocumentPeriodEndDate">
     	          2021
	</ix:nonNumeric>
</ix:nonNumeric>
```
However, the parser should now be able to handle (in theory indefinitely deep) nesting, since it now loops recursively over all iXBRL tags and extracts all text that is encapsulated by the tag and is not part of any XML code.
